state-sync/src/executor_proxy.rs

```diff
@@ -171,34 +198,53 @@ impl ExecutorProxyTrait for ExecutorProxy {
         known_version: u64,
         limit: u64,
         target_version: u64,
-    ) -> Result<TransactionListWithProof> {
+    ) -> Result<TransactionListWithProof, Error> {
         let starting_version = known_version
             .checked_add(1)
-            .ok_or_else(|| format_err!("Starting version has overflown!"))?;
+            .ok_or_else(|| Error::IntegerOverflow("Starting version has overflown!".into()))?;
         self.storage
             .get_transactions(starting_version, limit, target_version, false)
+            .map_err(|error| {
+                Error::UnexpectedError(format!("Failed to get transactions from storage {}", error))
+            })
     }
 
-    fn get_epoch_change_ledger_info(&self, epoch: u64) -> Result<LedgerInfoWithSignatures> {
+    fn get_epoch_change_ledger_info(&self, epoch: u64) -> Result<LedgerInfoWithSignatures, Error> {
         let next_epoch = epoch
             .checked_add(1)
-            .ok_or_else(|| format_err!("Next epoch has overflown!"))?;
-        self.storage
-            .get_epoch_ending_ledger_infos(epoch, next_epoch)?
+            .ok_or_else(|| Error::IntegerOverflow("Next epoch has overflown!".into()))?;
```

Comment:
This is kinda why I mentioned earlier about the `&'static str` vs `String` so you wouldn't have to do all the `into()` calls.

Reply:
True, although there are other situations where you have to convert a `String` into a `str` for these errors, so I've just been consistent with what already have.